dotnet/src/SemanticKernel.Abstractions/Orchestration/FunctionResult.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.SemanticKernel.Orchestration;
+
+/// <summary>
+/// Function result after execution.
+/// </summary>
+public sealed class FunctionResult
+{
+    /// <summary>
+    /// Additional data from model.
+    /// </summary>
+    public IReadOnlyCollection<ModelResult> ModelResults { get; internal set; } = Array.Empty<ModelResult>();
+
+    /// <summary>
+    /// Function result object.
+    /// </summary>
+    internal object? Value { get; private set; } = null;
+
+    /// <summary>
+    /// Instance of <see cref="SKContext"/> to pass in function pipeline.
+    /// </summary>
+    internal SKContext Context { get; private set; }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="FunctionResult"/> class.
+    /// </summary>
+    /// <param name="context">Instance of <see cref="SKContext"/> to pass in function pipeline.</param>
+    public FunctionResult(SKContext context)
+    {
+        this.Context = context;
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="FunctionResult"/> class.
+    /// </summary>
+    /// <param name="context">Instance of <see cref="SKContext"/> to pass in function pipeline.</param>
+    /// <param name="value">Function result object.</param>
+    public FunctionResult(SKContext context, object? value)
+    {
+        this.Value = value;
+        this.Context = context;
+    }
+
+    /// <summary>
+    /// Returns function result value.
+    /// </summary>
+    /// <typeparam name="T">Target type for result value casting.</typeparam>
+    /// <exception cref="InvalidCastException">Thrown when it's not possible to cast result value to <typeparamref name="T"/>.</exception>
+    public T? GetValue<T>()
```

Comment:
Should this ne named `GetValueOrDefault` or should it return null if the `Value` is null?

Reply:
Regarding naming, I would like to avoid long name for this method, as it should be widely used, so keeping it as short as possible will simplify the usage. 
In terms of returning null - it will return null in case `Value` is reference type, since `default` value for reference types is `null`. Also, we cannot return null in all cases, since `T` can be non-nullable value type, like `int` for example.